### ISSUE TITLE
Yields section requires wrong type annotation

### ISSUE THREAD

**Issue by haakonvt** [jsh9/pydoclint#75]
For a generator (function) like the following, according to the Google style docs, the annotation in the `Yields:` section should be annotated as follows:

```py
from typing import Generator


def foo(n: int) -> Generator[int, None, None]:
    """Description

    Args:
        n (int): Description

    Yields:
        int: Description
    """
    yield from range(n)
```

This throws an error when I try to run it through `pydoclint`:
```
4: DOC404: Function `foo` yield type(s) in docstring not consistent with the return annotation.
  Return annotation types: ['Generator[int, None, None]']; docstring return section types: ['int']
```

I.e., it wants:
```
    Yields:
        Generator[int, None, None]: Description
```

Is this a bug?

**Comment by jsh9** [comment]
Hi @haakonvt , thank you for discovering this!  It's indeed a bug, and it's fixed in [v0.2.3](https://github.com/jsh9/pydoclint/releases/tag/0.2.3).

**Comment by jsh9** [comment]
Hi @erlendvollset , thank you for providing those examples!

My fix addresses the 1st combination:

```python
def foo(n: int) -> Generator[int, None, None]:
    """Description

    Args:
        n (int): Description

    Yields:
        int: Description
    """
    yield from range(n)
```

The 2nd combination you mentioned above is a very interesting idea.  I'll need to think about how to implement it though.  (And if it "interferes" with existing violation checking logic, I may choose not to implement it.)

As to the 3rd and the 4th combinations, please read this note that I wrote: https://jsh9.github.io/pydoclint/notes_generator_vs_iterator.html

**Comment by haakonvt** [comment]
> Hi @haakonvt , thank you for discovering this! It's indeed a bug, and it's fixed in [v0.2.3](https://github.com/jsh9/pydoclint/releases/tag/0.2.3).

Great, thanks (again x3) 😄

**Comment by haakonvt** [comment]
@jsh9 Just updated, worked for all cases except one which reports a `DOC404`:
```
5: DOC404: Method `Foo.create` yield type(s) in docstring not consistent with the return
  annotation. The yield type (the 0th arg in Generator[...]): Generator[tuple[float, ...], None, None];
  docstring "yields" section types: tuple[float, ...]
```

The code that triggered it:
```py
from typing import Generator

class Foo:
    def create(self, n: int) -> Generator[tuple[float, ...], None, None]:
        """Foo

        Args:
            n (int): Description.

        Yields:
            tuple[float, ...]: Description.
        """
        yield from ((*self.bar, i) for i in range(self.baz))
```

**Comment by jsh9** [comment]
Yep, I just realized that issue too.  I fixed it in [v0.2.4](https://github.com/jsh9/pydoclint/releases/tag/0.2.4).

**Comment by jsh9** [comment]
Hi @erlendvollset and @haakonvt :

I released v0.3.0, which allowed this combination in your example:

```python
def foo3(n: int) -> Iterator[int]:
    """Description

    Args:
        n (int): Description

    Yields:
        int: Description
    """
    yield from range(n)
```

See this issue (https://github.com/jsh9/pydoclint/issues/76) for more details.
